pallets/stateful-storage/src/lib.rs

```diff
@@ -274,20 +277,84 @@ pub mod pallet {
 			Ok(())
 		}
 	}
+}
 
-	impl<T: Config> Pallet<T> {
-		pub fn get_itemized_page(
-			msa_id: MessageSourceId,
-			schema_id: SchemaId,
-		) -> Page<T::MaxItemizedPageSizeBytes> {
-			let storage_key = &schema_id.encode()[..];
-			let keys = vec![storage_key.to_vec()];
-			let page_response =
-				StatefulChildTree::try_read::<Page<T::MaxItemizedPageSizeBytes>>(&msa_id, &keys)
-					.map_or_else(|_| Page::default(), |page| page.unwrap_or_default());
-			page_response
+impl<T: Config> Pallet<T> {
+	/// This function returns all the pages associated with `msa_id` and `schema_id` params
+	/// and support both types of storage types
+	/// Itemized - For this storage it parses all the items and returns each item with its index
+	/// Paginated - For this storage it finds all the pages and return each page with its pageId
+	///
+	/// [Warning] since this function iterates over all the potential keys it should never called
+	/// from runtime.
+	pub fn get_pages(msa_id: MessageSourceId, schema_id: SchemaId) -> Vec<PageResponse> {
```

Comment:
This does not return any error and just ignores any if happened and return only valid items.
Let me know if you think it makes sense to expose errors also.

Reply:
Imo: its ok like this, if request is legit! then they would expect results else empty , we follow this pattern at quite a few places, but not opposed to errors if they will actually be meaningful

Reply:
I agree. No need to return an error unless there's something meaningful or critical.